packages/search-input/src/SearchInput/SearchInput.types.ts

```diff
@@ -0,0 +1,69 @@
+import {
+  ChangeEventHandler,
+  ComponentPropsWithoutRef,
+  FormEventHandler,
+} from 'react';
+
+import { AriaLabelProps } from '@leafygreen-ui/a11y';
+import { DarkModeProps } from '@leafygreen-ui/lib';
+
+export const State = {
+  Unset: 'unset',
+  Loading: 'loading',
+} as const;
+
+export type State = typeof State[keyof typeof State];
+
+export const SizeVariant = {
+  Small: 'small',
+  Default: 'default',
+  Large: 'large',
+} as const;
+
+export type SizeVariant = typeof SizeVariant[keyof typeof SizeVariant];
+
+interface BaseSearchInputProps
+  extends DarkModeProps,
+    Omit<ComponentPropsWithoutRef<'form'>, 'onChange'> {
+  /**
+   * The current state of the SearchInput. This can be none, or loading.
+   */
+  state?: State;
+
+  /**
+   * Determines the font size and padding.
+   */
+
+  sizeVariant?: SizeVariant;
```

Comment:
why is this `sizeVariant` when all of the other props with a similar behavior are `size`?

Reply:
This was ported over from `TextInput`. @shaneeza Do you remember why we used this prop name?

Reply:
yeah i initially added `sizeVariant` cause `input` has a `size` attribute but looking back now I could have just omitted `size`.

Reply:
This component inherits from `form` anyway, so doesn't have a native `size` attr. 
In general though, I'm in favor of overriding the `size` attribute. I'd be surprised if anyone uses it, and it will be ignored in favor of any explicit css `width` rule